### PR TITLE
Add support for einsum pattern bqnc,bnchw->bqnhw to unblock detr's segmentation test

### PR DESCRIPTION
**Pull request by kamalrajkannan78**
## Summary 

- Detr's segmentation Model needs support for einsum pattern **bqnc,bnchw->bqnhw** because of [this](https://github.com/huggingface/transformers/blob/5d7739f15a6e50de416977fe2cc9cb516d67edda/src/transformers/models/detr/modeling_detr.py#L1801C9-L1801C107). This PR adds support for it.

## Logs

- After adding support - [einsum.log](https://github.com/user-attachments/files/18824454/feb17_new_einsum_f.log)

### CODE REVIEW

File: forge/test/mlir/operators/eltwise_binary/test_eltwise_binary.py [hunk 1]
```diff
@@ -11,6 +11,38 @@
 from forge.verify.config import VerifyConfig
 
 
+@pytest.mark.xfail(reason="Tensor Mismatch at runtime")
+@pytest.mark.parametrize(
```
Comment (ashokkumarkannan1) [inline forge/test/mlir/operators/eltwise_binary/test_eltwise_binary.py:15]: Can we have `equation` also a parametrised input? It will be  easy to add other equation tests too in the future.

File: forge/test/mlir/operators/eltwise_binary/test_eltwise_binary.py [hunk 2]
```diff
@@ -11,6 +11,38 @@
...
+    "input_1, input_2",
+    [
+        ((1, 100, 8, 32), (1, 8, 32, 14, 20)),
+        ((1, 50, 8, 32), (1, 8, 32, 16, 16)),
```
Comment (ashokkumarkannan1) [inline forge/test/mlir/operators/eltwise_binary/test_eltwise_binary.py:19]: Two tests are fine as we are not doing multi-dim tests here. Let one test have batch size greater than 1, and shapes having odd numbers and not a factors of 32(TILE_DIM).

File: forge/test/mlir/operators/eltwise_binary/test_eltwise_binary.py [hunk 3]
```diff
@@ -11,6 +11,38 @@
 from forge.verify.config import VerifyConfig
 
 
+@pytest.mark.xfail(reason="Tensor Mismatch at runtime")
```
Comment (ashokkumarkannan1) [inline forge/test/mlir/operators/eltwise_binary/test_eltwise_binary.py:14]: Can we also have the exact PCC value for future reference?

**Review by ashokkumarkannan1 — CHANGES_REQUESTED** [review]

**Review by ashokkumarkannan1 — APPROVED** [review]
LGTM!!! Thanks for the PR @kamalrajkannan78

File: forge/test/mlir/operators/eltwise_binary/test_eltwise_binary.py [hunk 4]
```diff
@@ -11,6 +11,48 @@
 from forge.verify.config import VerifyConfig
 
 
+import pytest
+import torch
+
+
+@pytest.mark.parametrize(
+    "einsum_pattern, input_1, input_2",
```
Comment (vkovinicTT) [inline forge/test/mlir/operators/eltwise_binary/test_eltwise_binary.py:19]: just rename `input_1` and `input_2` into `shape_1` and `shape_2` so it's clearer what 2 of those are :))

**Review by vkovinicTT — CHANGES_REQUESTED** [review]

**Comment by vkovinicTT** [comment]
Is the attached log after adding support in tvm or before?

**Comment by vkovinicTT** [comment]
> > Is the attached log after adding support in tvm or before? Please also attach corresponding tvm PR.
> 
> Yes, the attached log corresponds to the state after adding support in TVM.
oh sorry you even wrote: "After adding support"... my bad

**Review by vkovinicTT — APPROVED** [review]
